src/Pact/Native.hs

```diff
@@ -438,10 +438,18 @@ langDefs =
      ["(typeof \"hello\")"] "Returns type of X as string."
     ,setTopLevelOnly $ defRNative "list-modules" listModules
      (funType (TyList tTyString) []) [] "List modules available for loading."
-    ,defRNative (specialForm Yield) yield (funType yieldv [("OBJECT",yieldv)])
-     [LitExample "(yield { \"amount\": 100.0 })"]
-     "Yield OBJECT for use with 'resume' in following pact step. The object is similar to database row objects, in that \
-     \only the top level can be bound to in 'resume'; nested objects are converted to opaque JSON values."
+
+    ,defRNative (specialForm YieldForm) yield
+     (funType yieldv [("object",yieldv)] <>
+      funType yieldv [("target-chain",tTyString),("object", yieldv)])
+     [ LitExample "(yield { \"amount\": 100.0 })"
+     , LitExample "(yield \"Testnet00/2\" { \"amount\": 100.0 })"
+     ]
+     "Yield OBJECT for use with 'resume' in following pact step on chain TARGET-CHAIN. If no chain is specified, \
+     \the following pact step must execute on the current chain. The object is similar to database row objects, \
+     \in that only the top level can be bound to in 'resume'; nested objects are converted to opaque JSON \
+     \values."
+
```

Comment:
I've added some space here for legibility. The wall of text makes things hard to read. Will revert if there is outcry

Reply:
- The "opaque JSON" from old docs is obsolete.
- Doco'ing LISP optional args proceeds from default to non-default case canonically so suggest "Yield OBJECT ... following pact step. With optional argument TARGET-CHAIN, target subsequent step to execute on targeted chain using automated SPV endorsement-based dispatch."

Reply:
Added